source/kraken/maintenance_worker.cpp

```diff
@@ -375,6 +378,10 @@ void MaintenanceWorker::handle_rt_in_batch(const std::vector<AmqpClient::Envelop
         }
         LOG4CPLUS_TRACE(logger, "received entity: " << feed_message.DebugString());
         for (const auto& entity : feed_message.entity()) {
+            auto res = applied_visited_id.insert(entity.id());
+            if (!res.second) {
+                continue;
+            }
```

Comment:
```suggestion
            auto res = applied_visited_id.insert(entity.id());
            // an newer disruption with the same id has already been seen, so we can ignore this one
            if (!res.second) {
                continue;
            }
```

I am a bit worried about this  "ignore disruptions whose id has already been seen".

- do we receive kirin disruptions with the same id ? When this happens are we sure we only need to take into account the last one ? Can we have two disruptions with the same id but that affect different vj ? poke @pbougue 
- what about chaos disruptions ? In particular, when we cancel a disruption, don't we get the same id twice ?

Reply:
This may deserve a code-comment, as we both asked ourselves the question :smiley:

* For kirin this is guaranteed that entity ID is the same only when the VJ is the same, and that taking only the last is valid.
Details: actually it's not so true with parallelism on kirin, but the guarantee is that it's gonna be as good as it was (nothing else can be used to decide which message is the last one currently).
:bulb: This may deserve a comment in https://github.com/hove-io/chaos-proto/blob/master/kirin_proto_doc.rs ?

* For chaos, this was discussed with chaos team and seemed OK, but I don't know about that particular case, I let @xlqian reply if he knows.

Reply:
Allright if this has been checked, it sounds good to me.
I do agree that some comments (here and in the proto doc) explaining this may be useful.

> For kirin this is guaranteed that entity ID is the same only when the VJ is the same
same VJ *and* same date ?

Reply:
Yes, same VJ and same date :+1: 

:white_check_mark: I'll try to add a little something into kirin_proto_doc.rs

Reply:
@pbench Now I'm having second thought with your comment and I think I made a mistake after thinking it through... 
@pbougue I'm going remove this trick(reversing the vector) in this PR and open up another PR to tackle this problem, so that in case i messed it up, we don't have to revert the whole thing

Reply:
We also have to reverse the order of the entities in the message to be safe that we still have the same result (although it would be weird for someone to send multiple time the same entity in the same message).

Reply:
After discussion: this was reverted from this PR because of uncertainty around chaos.
Tracked in JIRA https://navitia.atlassian.net/browse/NAV-1878